<robot name="jibo">

    <gazebo>
        <plugin name="gazebo_ros_control" filename="libgazebo_ros_control.so">
        </plugin>
    </gazebo>
	<!-- * * * Link Definitions * * * -->
    <link name="base_link">
 	    <inertial>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <mass value="2.0" />
            <inertia ixx="0.003224" ixy="0.0" ixz="0.0" iyy="0.003224" iyz="0.0" izz="0.0064"/>
        </inertial>
        <collision>
            <origin xyz="0 0 -0.006" rpy="0 0 0"/>
            <geometry>
                <cylinder radius="0.08" length="0.012"/>
            </geometry>
        </collision>
        <visual>
            <origin rpy="0.0 0 0" xyz="0 0 0"/>
            <geometry>
                <mesh filename="package://jibo_description/models/jibo/meshes/jibo_final_parts_base_v3.dae"/>
            </geometry>
        </visual>
	</link>

    <gazebo reference="base_link">
        <kp>1000000.0</kp>
        <kd>0.0</kd>
        <mu1>10.0</mu1>
        <mu2>10.0</mu2>
    </gazebo>

    <link name="waist_link">
 	    <inertial>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <mass value="0.5" />
            <inertia ixx="0.0011625" ixy="0.0" ixz="0.0" iyy="0.0011625" iyz="0.0" izz="0.002025"/>
        </inertial>
        <collision>
            <origin xyz="0 0 0.03" rpy="0 0 0"/>
            <geometry>
                <cylinder radius="0.09" length="0.06"/>
            </geometry>
        </collision>
		<visual>
            <origin rpy="0.0 0 0" xyz="0 0 0"/>
            <geometry>
                <mesh filename="package://jibo_description/models/jibo/meshes/jibo_final_parts_waist_v3.dae"/>
            </geometry>
        </visual>
	</link>

    <joint name="base_waist_joint" type="continuous">
    	<parent link="base_link"/>
    	<child link="waist_link"/>
    	<origin xyz="0 0 0" rpy="0 0 0"/>
        <limit effort="1.0" velocity="1.0"/>
        <axis xyz="0 0 1"/>
	</joint>

    <transmission name="tran1">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="base_waist_joint">
            <hardwareInterface>EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="motor1">
            <hardwareInterface>EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>


    <link name="body_link">
 	    <inertial>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <mass value="0.5" />
            <inertia ixx="0.00166354166667" ixy="0.0" ixz="0.0" iyy="0.00166354166667" iyz="0.0" izz="0.002025"/>
        </inertial>
        <collision>
            <origin xyz="0 0 0.06" rpy="0 0 0"/>
            <geometry>
                <cylinder radius="0.09" length="0.125"/>
            </geometry>
        </collision>
		<visual>
            <origin rpy="0.0 0 0" xyz="0 0 0"/>
            <geometry>
                <mesh filename="package://jibo_description/models/jibo/meshes/jibo_final_parts_body_v3.dae"/>
            </geometry>
        </visual>
	</link>

    <joint name="waist_body_joint" type="continuous">
    	<parent link="waist_link"/>
    	<child link="body_link"/>
    	<origin xyz="0 0.006 0.052" rpy="-0.263 0 0"/>
        <limit effort="1.0" velocity="1.0"/>
        <axis xyz="0 0 1"/>
	</joint>

    <transmission name="tran2">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="waist_body_joint">
            <hardwareInterface>EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="motor2">
            <hardwareInterface>EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>

    <link name="head_link">
 	    <inertial>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <mass value="0.1" />
            <inertia ixx="0.000625" ixy="0.0" ixz="0.0" iyy="0.000625" iyz="0.0" izz="0.000625"/>
        </inertial>
        <collision>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <geometry>
                <sphere radius="0.11"/>
            </geometry>
        </collision>
		<visual>
            <origin rpy="0.0 0 0" xyz="0 0 0"/>
            <geometry>
                <mesh filename="package://jibo_description/models/jibo/meshes/jibo_final_parts_head_v5.dae"/>
            </geometry>
        </visual>
	</link>

    <joint name="body_head_joint" type="continuous">
    	<parent link="body_link"/>
    	<child link="head_link"/>
    	<origin xyz="0 0.0 0.19" rpy="-0.4 0 0"/>
        <limit effort="1.0" velocity="1.0"/>
        <axis xyz="0 0 1"/>
	</joint>

    <transmission name="tran3">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="body_head_joint">
            <hardwareInterface>EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="motor3">
            <hardwareInterface>EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>



    <link name="camera_link">
        <!-- Inertial for gazebo to take it into account, Put 1gr weight -->
        <inertial>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <mass value="0.001" />
            <inertia ixx="9.1875e-09" ixy="0.0" ixz="0.0" iyy="9.1875e-09" iyz="0.0" izz="1.378125e-08"/>
        </inertial>
		<visual>
            <origin rpy="0.0 0 0" xyz="0 0 0"/>
            <geometry>
                <box size="0.0005 0.0005 0.0005"/>
            </geometry>
            <material name="green"/>
        </visual>
	</link>

    <gazebo reference="camera_link">
        <material>Gazebo/Green</material>
    </gazebo>

    <joint name="head_camera_joint" type="fixed">
        <parent link="head_link"/>
        <child link="camera_link"/>
        <origin xyz="0 0.013 0.1" rpy="0.4 0 0"/>
    </joint>


    <!-- camera -->
  <gazebo reference="camera_link">
    <sensor type="camera" name="camera1">
      <update_rate>10.0</update_rate>
      <camera name="head">

          <pose>0 0 0 0 0 1.57</pose>

        <horizontal_fov>1.3962634</horizontal_fov>
        <image>
          <width>400</width>
          <height>400</height>
          <format>R8G8B8</format>
        </image>
        <clip>
          <near>0.01</near>
          <far>100</far>
        </clip>
        <noise>
          <type>gaussian</type>
          <stddev>0.007</stddev>
        </noise>
      </camera>
      <plugin name="camera_controller" filename="libgazebo_ros_camera.so">
        <alwaysOn>true</alwaysOn>
        <updateRate>0.0</updateRate>
        <cameraName>jibo/camera1</cameraName>
        <imageTopicName>image_raw</imageTopicName>
        <cameraInfoTopicName>camera_info</cameraInfoTopicName>
        <frameName>camera_link</frameName>
        <hackBaseline>0.07</hackBaseline>
        <distortionK1>0.0</distortionK1>
        <distortionK2>0.0</distortionK2>
        <distortionK3>0.0</distortionK3>
        <distortionT1>0.0</distortionT1>
        <distortionT2>0.0</distortionT2>
      </plugin>
    </sensor>
  </gazebo>


</robot>